pkg_resources/__init__.py

```diff
@@ -2699,15 +2699,15 @@ def __init__(
         self.dist = dist
 
     def __str__(self):
-        s = "%s = %s" % (self.name, self.module_name)
+        s = f"{self.name} = {self.module_name}"
         if self.attrs:
             s += ':' + '.'.join(self.attrs)
         if self.extras:
-            s += ' [%s]' % ','.join(self.extras)
+            s += ' [{}]'.format(','.join(self.extras))
         return s
 
     def __repr__(self):
-        return "EntryPoint.parse(%r)" % str(self)
+        return f"EntryPoint.parse({str(self)!r})"
```

Comment:
Not an automated change, but this can be simplified further

```suggestion
        return f"EntryPoint.parse('{self}')"
```

Reply:
Detection at least can be automated:
```console
$ grep -R '{str(' --exclude-dir=_distutils --exclude-dir=_vendor
setuptools/tests/test_sdist.py:                                depends=[{str(dep_path)!r}]
pkg_resources/__init__.py:        return f"EntryPoint.parse({str(self)!r})"
pkg_resources/__init__.py:                raise _packaging_version.InvalidVersion(f"{str(ex)} {info}") from None
pkg_resources/__init__.py:            {str(ex)}\n{notes}
pkg_resources/__init__.py:        return f"Requirement.parse({str(self)!r})"
$ 
```

Reply:
Actually, I have seen `str(self)!r}` elsewhere, especially in _distutils_, and I believe this is on purpose. From the [`repr()`](https://docs.python.org/fr/dev/library/functions.html#repr) documentation:
> For many types, this function makes an attempt to return a string that would yield an object with the same value when passed to [eval()](https://docs.python.org/fr/dev/library/functions.html#eval);

We precisely want an expression that yields `str(self)` when executed. Of course, `{str(self)!r})` is equivalent to `'{self}'`, but the former expression conveys a specific intent.

Reply:
Yeah it's not the same as a standalone `str` or `repr` in a fstring. The former being truly redundant, and the later changeable to `!r`